### PR TITLE
add resource and description fields for TypeDef

### PR DESCRIPTION
**Pull request by ss2165**
almost parity with `OpDef`

### CODE REVIEW

File: src/resource.rs [hunk 1]
```diff
@@ -344,18 +348,24 @@ impl Resource {
     }
 
     /// Add an exported type to the resource.
-    pub fn add_type(&mut self, ty: TypeDef) {
+    pub fn add_type(&mut self, ty: TypeDef) -> Result<(), String> {
+        if ty.resource != ResourceId::default() {
```
Comment (aborgna-q) [inline src/resource.rs:352]: This never updates the empty `ty.resource`?

Maybe we should use an `Option<SmolStr>` to flag when the definition hasn't been attached to a resource, and overwrite it before inserting it to `self.types`? (and do the same for OpDef)

I just checked, and `SmolStr` works well with NPO (it takes 24 bytes both with and without the `Option`).

Comment (ss2165) [reply src/resource.rs]: sorry yeah not updating the resource was an oversight, took your suggestion

**Review by aborgna-q — APPROVED** [review]
We could do the same Option<> thing for OpDef to keep the symmetry.
But feel free to requeue the merge, we can do it later too.

**Comment by ss2165** [comment]
> We could do the same Option<> thing for OpDef to keep the symmetry. 
I have done already?

**Pull request merged**